server/lib/pbench/test/conftest.py

```diff
@@ -0,0 +1,126 @@
+"""
+
+
+@pytest.fixture(scope="session", autouse=True)
+def setup(request, pytestconfig):
+    """Test package setup for pbench-server"""
+
+    # Create a single temporary directory for the "/srv/pbench" and
+    # "/opt/pbench-server" directories.
+    tmp = tempfile.TemporaryDirectory(suffix=".d", prefix="pbench-server-unit-tests.")
+    tmp_d = Path(tmp.name)
+
+    srv_pbench = tmp_d / "srv" / "pbench"
+    pbench_tmp = srv_pbench / "tmp"
+    pbench_tmp.mkdir(parents=True, exist_ok=True)
+    pbench_logs = srv_pbench / "logs"
+    pbench_logs.mkdir(parents=True, exist_ok=True)
+    pbench_recv = srv_pbench / "pbench-move-results-receive" / "fs-version-002"
+    pbench_recv.mkdir(parents=True, exist_ok=True)
+
+    opt_pbench = tmp_d / "opt" / "pbench-server"
+    pbench_bin = opt_pbench / "bin"
+    pbench_bin.mkdir(parents=True, exist_ok=True)
+    pbench_cfg = opt_pbench / "lib" / "config"
+    pbench_cfg.mkdir(parents=True, exist_ok=True)
+    pbench_archive = srv_pbench / "archive" / "fs-version-001"
+    pbench_archive.mkdir(parents=True, exist_ok=True)
+
+    # "Install" the default server configuration file.
+    shutil.copyfile(
+        "./server/lib/config/pbench-server-default.cfg",
+        (pbench_cfg / "pbench-server-default.cfg"),
+    )
+
+    pbench_recv_tb = (
+        srv_pbench / "pbench-move-results-receive" / "fs-version" / "tarball-002"
+    )
+    pbench_recv_tb.mkdir(parents=True, exist_ok=True)
+    pbench_recv_bad_tb = (
+        srv_pbench / "pbench-move-results-receive" / "fs-version" / "bad_tarball-002"
+    )
+    pbench_recv_bad_tb.mkdir(parents=True, exist_ok=True)
+    shutil.copyfile(
+        "./server/lib/pbench/test/test-fixtures/upload/log.tar.xz.md5",
+        (pbench_recv_bad_tb / "bad_log.tar.xz.md5"),
+    )
+    shutil.copyfile(
+        "./server/lib/pbench/test/test-fixtures/upload/log.tar.xz",
+        (pbench_recv_tb / "log.tar.xz"),
+    )
+    shutil.copyfile(
+        "./server/lib/pbench/test/test-fixtures/upload/log.tar.xz.md5",
+        (pbench_recv_tb / "log.tar.xz.md5"),
+    )
+
+    pbench_recv_badtb_tb = (
+        srv_pbench / "pbench-move-results-receive" / "fs-version" / "mix_tarball-002"
+    )
+    pbench_recv_badtb_tb.mkdir(parents=True, exist_ok=True)
+    shutil.copyfile(
+        "./server/lib/pbench/test/test-fixtures/upload/log.tar.xz.md5",
+        (pbench_recv_badtb_tb / "bad_log.tar.xz.md5"),
+    )
+    shutil.copyfile(
+        "./server/lib/pbench/test/test-fixtures/upload/log.tar.xz",
+        (pbench_recv_badtb_tb / "log.tar.xz"),
+    )
+    shutil.copyfile(
+        "./server/lib/pbench/test/test-fixtures/upload/log.tar.xz.md5",
+        (pbench_recv_badtb_tb / "log.tar.xz.md5"),
+    )
+
+    cfg_file = pbench_cfg / "pbench-server.cfg"
+    cfg_file.write_text(server_cfg_tmpl.format(tmp=tmp_d))
+    pytestconfig.cache.set("TMP", str(tmp_d))
+    pytestconfig.cache.set("_PBENCH_SERVER_CONFIG", str(cfg_file))
+
+    request.addfinalizer(lambda: tmp.cleanup())
```

Comment:
Since this is a fixture, another option would be to `yield` here and let it return to `tmp.cleanup()` after the test method finishes ... but this is fine.

Reply:
This is what it was doing earlier.